Chess/include/Board.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <iostream>
+#include <vector>
+#include <string>
+//#include "Piece.h"
+
+using std::string;
+
+class Piece;
```

Comment:
Use forward declarations: instead of including the entire header file for Piece and King, you can use forward declarations. This can help reduce compilation dependencies and improve build times.